### PR TITLE
AMQP-473: RabbitManagementTemplate

### PR DESCRIPTION
**Pull request by garyrussell**
TODO: Docs

### CODE REVIEW

**Comment by artembilan** [comment]
I don't see issues with this PR.
Just only the point to the M1 of Hop. If we are going to release this with our 1.5.0.M1, of course...

**Comment by garyrussell** [comment]
Yes, it's ok for us to rely on a milestone in our M1.

**Comment by artembilan** [comment]
Good. So, go ahead, please, with docs and I'll merge it!

**Comment by garyrussell** [comment]
Pushed

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+		}
+		return convertedQueues;
+	}
+
+	private Queue convert(QueueInfo qi) {
+		return new Queue(qi.getName(), qi.isDurable(), qi.isExclusive(), qi.isAutoDelete(),
+				qi.getArguments());
+	}
+
+	private List<Exchange> converteExchangeList(List<ExchangeInfo> exchanges) {
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java]: **converte** ? It is valid word, buy the way: http://en.wikipedia.org/wiki/Galician_language
But I hope we prefer English :smile: 
Will fix on merge.

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+		else if (ei.getType().equals("topic")) {
+			return new TopicExchange(ei.getName(), ei.isDurable(), ei.isAutoDelete(), ei.getArguments());
+		}
+		else {
+			return null;
+		}
+
+	}
+
+	private List<Binding> converteBindingList(List<BindingInfo> bindings) {
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java]: DITO

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+	 * @param uri the uri.
+	 * @param username the user.
+	 * @param password the password.
+	 */
+	public RabbitManagementTemplate(String uri, String username, String password) {
+		try {
+			this.rabbitClient = new Client(uri, username, password);
+		}
+		catch (Exception e) {
+			throw new AmqpException(e);
```
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java]: Let's discuss this part.
Really not sure that we should wrap any underlying exception to such a specific `AmqpException`.
What is the concern do rethrow as is? Just make it more friendly for end-users?

File: spring-rabbit/src/test/java/org/springframework/amqp/rabbit/test/BrokerRunning.java [hunk 4]
```diff
@@ -198,6 +219,13 @@ public Statement apply(Statement base, Description description) {
 				Assume.assumeTrue(brokerOffline.get(port));
 			}
 
+			if (this.management) {
+				Client client = new Client("http://localhost:15672/api/", "guest", "guest");
+				if (!client.alivenessTest("/")) {
+					throw new RuntimeException("Aliveness test failed for locahost:15672 guest/quest; "
```
Comment (artembilan) [inline spring-rabbit/src/test/java/org/springframework/amqp/rabbit/test/BrokerRunning.java]: typo: s/locahost/localhost

**Comment by artembilan** [comment]
Don't you add `What's New` note intentionally?
It's not the first time, when you have missed that...
Should I do that on merge or share with you the polishing commit?
Thanks

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java (hunk 1 above)
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java]: typo

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java (hunk 3 above)
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java]: I guess I just don't like checked exceptions (and there are two of them).

**Comment by garyrussell** [comment]
I simply forgot about what's new :frowning:

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java (hunk 3 above)
Comment (artembilan) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitManagementTemplate.java]: OK. Nevermind. Accepted

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-amqp/commit/69836a657a8838fddc429a983c3b3fe6da9d360a

**Pull request closed without merging**